test/LanguageServer.Engine.Tests/TestProject.cs

```diff
@@ -0,0 +1,386 @@
+        /// </param>
+        /// <param name="relativePathSegments">
+        ///     The project file's relative (to the test deployment directory) path segments.
+        /// </param>
+        /// <returns>
+        ///     The full path to the project file.
+        /// </returns>
+        public static string GetProjectDirectory(Type testClassType, params string[] relativePathSegments)
+        {
+            if (relativePathSegments == null)
+                throw new ArgumentNullException(nameof(relativePathSegments));
+
+            string projectFileName = GetProjectFile(testClassType, relativePathSegments);
+
+            return Path.GetDirectoryName(projectFileName);
+        }
+
+        /// <summary>
+        ///     Get the deployment directory for the specified test class.
+        /// </summary>
+        /// <param name="testClassType">
+        ///     The test class <see cref="Type"/>.
+        /// </param>
+        /// <returns>
+        ///     The test deployment directory.
+        /// </returns>
+        static string GetTestDeploymentDirectory(Type testClassType)
+        {
+            if (testClassType == null)
+                throw new ArgumentNullException(nameof(testClassType));
+
+            string testAssemblyFile = testClassType.Assembly.Location;
+            
+            return Path.GetDirectoryName(testAssemblyFile);
+        }
+    }
+
+    /// <summary>
+    ///     An MSBuild project, loaded for a test.
+    /// </summary>
+    /// <param name="ProjectXml">
+    ///     An <see cref="XmlDocumentSyntax"/> containing the project XML.
+    /// </param>
+    /// <param name="XmlLocations">
+    ///     The <see cref="XmlLocator"/> for the project XML.
+    /// </param>
+    /// <param name="TextPositions">
+    ///     The <see cref="TextPositions"/> for the project text.
+    /// </param>
+    public record class TestProjectXml(XmlDocumentSyntax ProjectXml, XmlLocator XmlLocations, TextPositions TextPositions);
+
+    /// <summary>
+    ///     An MSBuild project, loaded for a test.
+    /// </summary>
+    /// <param name="MSBuildProject">
+    ///     The underlying MSBuild <see cref="Project"/>.
+    /// </param>
+    /// <param name="ObjectLocations">
+    ///     The <see cref="MSBuildObjectLocator"/> for the project.
+    /// </param>
+    /// <param name="ProjectXml">
+    ///     An <see cref="XmlDocumentSyntax"/> containing the project XML.
+    /// </param>
+    /// <param name="XmlLocations">
+    ///     The <see cref="XmlLocator"/> for the project XML.
+    /// </param>
+    /// <param name="TextPositions">
+    ///     The <see cref="TextPositions"/> for the project text.
+    /// </param>
+    public record class TestProject(Project MSBuildProject, MSBuildObjectLocator ObjectLocations, XmlDocumentSyntax ProjectXml, XmlLocator XmlLocations, TextPositions TextPositions)
```

Comment:
There are too many parameters to fit on 1 line. Consider splitting like this:
```suggestion
    public record class TestProject(
        Project MSBuildProject,
        MSBuildObjectLocator ObjectLocations,
        XmlDocumentSyntax ProjectXml,
        XmlLocator XmlLocations,
        TextPositions TextPositions)
```

Reply:
What width do you have your editor set to? I don’t mind changing it but would be good to keep a good mental guideline (I’ll add it to editorconfig maybe?)

Reply:
I was looking from the github view. I have 2k screen and the line overflowed. That was the triggering factor)

Reply:
Fair call :)